tsinfer/inference.py

```diff
@@ -2011,6 +2020,17 @@ def post_process(
                 "Cannot find a virtual-root-like ancestor during preprocessing"
             )
 
+        if not keep_flanks and tables.sites.num_rows > 0:
+            logger.info("Removing topology in flanking regions with keep_intervals")
+            # So that the last site falls within a tree, we must add one to the
+            # site position (or simply extend to the end of the ts)
+            upper_cutoff = min(tables.sites.position[-1] + 1, tables.sequence_length)
```

Comment:
Need to mindful that creating these column arrays is not a constant time operation and we'll be working with very large tables. You've created two copies within a couple of lines here.

Reply:
Ah, very helpful comment, thanks. I thought that once we had the tables the arrays gave direct memory access, so weren't costly. In which case, what's the quickest way to find the last position? I also assumed that `tables.sites.num_rows` was the same cost as `ts.num_sites`, but I see that it might not be.

Reply:
Currently they create a new copy of the array each time you access. That's what ``ts.sites_position`` is for - **that** is a cheap view on the underlying memory.

Just store the array in a variable, ``pos = tables.sites.position`` to avoid needlessly recreating it.

Reply:
Ah, I thought `ts.sites_position` and tables.sites.position were equivalent. Thanks for the heads-up

Reply:
No need to create the array anyway, I just need the first and last position, and I have the tree sequence, so I can do `ts.sites_position[0]` and `ts.sites_position[-1]` (or `ts.site(0).position`, etc I guess)

Reply:
As a general rule so, if you have the ts it's *much* better to use one of those arrays than the equivalent table column.

Reply:
Useful to know, thanks. This is ready to merge, so.